### ISSUE TITLE
Make Home Assistant discovery prefix configurable

### ISSUE THREAD

**Issue by MintBlueJelly** [jomjol/AI-on-the-edge-device#3113]
### The Feature

Right now, the Homeassistant Discovery (using MQTT) uses the hard-coded discovery prefix "homeassistant". While this works with the HA MQTT integration with default settings, it will invariably fail when the discovery prefix has been customized to anything else (e.g. to "ha").

To fix this, please make the discovery prefix configurable and expose it as a new input field in the "Homeassistant Discovery (using MQTT)" section of the settings page.

**Comment by henrythasler** [comment]
I wonder if changing the discovery prefix is something a lot of people do on their setup or if that is some <1% super-rare edge-case. Maybe you could explain your motivation for requesting that feature.

**Comment by MintBlueJelly** [comment]
I can't comment on statistics, since I don't have the numbers. But I would tend to agree that it's fair to assume that customizing the discovery prefix is not something most users will do. Having said that, there are good reasons for making the discovery prefix configurable:

- The formal argument: The HA discovery prefix is designed to be customizable and [documented as such](https://www.home-assistant.io/integrations/mqtt/#discovery-messages). Solutions implementing HA MQTT Discovery should adhere to this design as a matter of course because otherwise they end up as partially incompatible.
- The practical argument: Prefix customization is required in any environment running multiple instances of Home Assistant with a shared MQTT broker (e.g. dedicated TEST/PROD instances). While it's possible to also have dedicated TEST/PROD instances of the MQTT broker, using a shared broker is more efficient.

In short, I believe following established examples (e.g. Zigbee2MQTT) and fully supporting discovery prefix customization would benefit not just my particular setup but this project as a whole.
